app/controllers/gigs_controller.rb

```diff
@@ -0,0 +1,92 @@
+class GigsController < ApplicationController
+  before_action :set_gig, only: [:show, :update, :destroy]
+
+  # GET /gigs
+  def index
+    brand_name = params[:brand_name]
+    creator_id = params[:creator_id]
+    @gigs = Gig.all
+    @gigs = @gigs.where(brand_name: brand_name) if brand_name
+    @gigs = @gigs.where(creator_id: creator_id) if creator_id
+
+    render json: @gigs
+
+    rescue ActiveRecord::RecordNotFound => e
+      logger.info(e)
+      render json: { error: e }, status: :not_found
+  end
+
+  # GET /gigs/1
+  def show
+    param! :include, String, in: ["gig_payment", "creator"]
+
+    render json: @gig, include: params[:include]
+
+    rescue RailsParam::InvalidParameterError => e
+      logger.info(e)
+      render json: { error: e }, status: :bad_request
+
+  end
+
+  # POST /gigs
+  def create
+    param! :creator_id, String, required: true
+    param! :brand_name, String, required: true
+
+    @gig = Gig.new(creator_id: params[:creator_id], brand_name: params[:brand_name])
+
+    if @gig.save
+      render json: @gig, status: :created, location: @gig
+    else
+      render json: @gig.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /gigs/1
+  def update
+    param! :creator_id, String, required: true
+    param! :brand_name, String, required: true
+
+    @gig = Gig.find(params[:id])
+    if @gig.update!(brand_name: params[:brand_name], creator_id: params[:creator_id])
+      render json: @gig
+    else
+      render json: @gig.errors, status: :unprocessable_entity
+    end
+
+    rescue ActiveRecord::RecordInvalid => e
+      logger.info(e)
+      render json: {error: e}, status: :not_found
+  end
+
+  # DELETE /gigs/1
+  def destroy
+    @gig.destroy
+  end
+
+  def set_completed
+    @gig = Gig.find(params[:id])
+    @gig.set_completed!
```

Comment:
I think the convention is to use just the verb for example `@gig.complete!` instead of `@gig.set_completed!`

Reply:
+you should catch AASM::InvalidTransition and look at what I told you via Slack about an atomic operation for both the AASM transition & gig attributes update